lib/src/http/common/request.dart

```diff
@@ -766,11 +766,11 @@ abstract class CommonRequest extends Object
       checkForTimeout();
       final maybeResponseCompleter = Completer<BaseResponse?>();
 
-      // Enforce a timeout threshold if set.
-      Timer? timeout;
-      if (timeoutThreshold != null) {
-        timeout = Timer(autoRetry.timeoutThreshold!, _timeoutRequest);
+      // Enforce a timeout threshold if specified.
+      if (timeoutThreshold == null && defaultTimeoutThreshold != null) {
+        timeoutThreshold = defaultTimeoutThreshold;
       }
+      final timeout = Timer(autoRetry.timeoutThreshold!, _timeoutRequest);
```

Comment:
```suggestion
      final timeout = Timer(timeoutThreshold!, _timeoutRequest);
```
Might need to do that, actually. And what if `timeoutThreshold == null && defaultTimeoutThreshold == null`? Then the `!` will throw, right? So maybe the timeout does need to be nullable.